src/core/server/saved_objects/service/lib/aggregations/validation.ts

```diff
@@ -181,11 +181,18 @@ const recursiveRewrite = (
 
     const nestedContext = childContext(context, key);
     const newKey = rewriteKey ? validateAndRewriteAttributePath(key, nestedContext) : key;
-    const newValue = rewriteValue
-      ? validateAndRewriteAttributePath(value, nestedContext)
-      : isPlainObject(value)
-      ? recursiveRewrite(value, nestedContext, [...parents, key])
-      : value;
+
+    let newValue = value;
+    if (rewriteValue) {
```

Comment:
nit -> to be more readable, will that work
```ts
if (rewriteValue) {
      newValue = validateAndRewriteAttributePath(value, nestedContext);
} else  if (isArray(value)) {
     newValue = value.map((v) => recursiveRewrite(v, nestedContext, parents));
} else if (isPlainObject(value)) {
     newValue = recursiveRewrite(value, nestedContext, [...parents, key]);
}
```

Reply:
done